client/ui-test/script-test.js

```diff
@@ -0,0 +1,129 @@
+			  if (tags[i].textContent == 'Dataset puppeteer test') {
+				var found = tags[i];
+				break;
+			  }
+			}
+			return found.parentElement.parentElement.parentElement.getAttribute('data-test-id');
+		});
+		await console.log("ID extracted: "+id+"\n");
+
+		//Visualisation
+		await console.log("Accessing to visualisation creation...");
+		await page.click('button[data-test-id="visualisation"]');
+		await console.log("Selecting pivot table option...");
+		await page.waitForSelector('li[data-test-id="button-pivot-table"]', {timeout:10000});
+		await page.click('li[data-test-id="button-pivot-table"]');
+		await console.log("Selecting dataset...");
+		await page.waitForSelector('label[data-test-id="dataset-menu"]+div', {timeout:10000});
+		await page.click('label[data-test-id="dataset-menu"]+div');
+		await page.waitForSelector('div[data-test-id="input-group"]>div>div>div:nth-of-type(2)', {timeout:10000});
+		await page.click('div[data-test-id="input-group"]>div>div>div:nth-of-type(2)');
+		await page.waitForSelector('label[data-test-id="categoryColumnInput"]+div', {timeout:10000});
+		await page.click('label[data-test-id="categoryColumnInput"]+div');
+		await page.waitForSelector('label[data-test-id="categoryColumnInput"]+div>div>div:nth-of-type(2)', {timeout:10000});
+		await page.click('label[data-test-id="categoryColumnInput"]+div>div>div:nth-of-type(2)');
+		await page.waitForSelector('input[name="categoryColumnInput"]', {timeout:10000});
+		await console.log("Selecting columns...");
+		await page.$eval('input[name="categoryColumnInput"]', el => el.value='c9');
+		await page.click('div[data-test-id="entity-title"]');
+		await console.log("Typing visualisation name...");
+		await page.type('input[data-test-id="entity-title"]', 'Visualisation of '+name);
+		await console.log("Saving visualisation...");
+		await page.click('button[data-test-id="save-changes"]');
+		await page.goto('http://t1.lumen.local:3030/library');
+		await page.waitForSelector('button[data-test-id="dashboard"]', {timeout:10000});
+		await console.log("Visualisation "+name+" was successfully created.\n");
+		//Dashboard
+		await console.log("Accessing to dashboard creation...");
+		await page.click('button[data-test-id="dashboard"]');
+		await console.log("Selecting visualisation...");
+		await page.waitForSelector('li[class^="listItem"]', {timeout:10000});
+		await page.click('li[class^="listItem"]');
+		await console.log("Typing dashboard name...");
+		await page.waitForSelector('div[data-test-id="dashboard-canvas-item"]', {timeout:10000});
+		await page.click('div[data-test-id="entity-title"]');
+		await page.type('input[data-test-id="entity-title"]', 'Dashboard of '+name);
+		await console.log("Saving dashboard...");
+		await page.click('button[data-test-id="save-changes"]');
+		await page.click('i[data-test-id="fa-arrow"]');
+		await console.log("Dashboard "+name+" was successfully created.\n");
+		await console.log("THE TEST WAS SUCCESSFUL");
```

Comment:
I don't see any asserts in the test. I can see failing if the "waitForSelector" timeouts. Any other causes for it to fail?